makefiles/testing.mak

```diff
@@ -107,20 +107,31 @@ units: $(CTAGS_TEST)
 	if ! test x$(TRAVIS)$(APPVEYOR)$(CIRCLECI) = x; then	\
 		SHOW_DIFF_OUTPUT=--show-diff-output;		\
 	fi;							\
+	if ! test x$(PYTHON) = x; then	\
+		PROG=$(PYTHON);		\
+		SCRIPT=$(srcdir)/misc/units.py;	\
+	else	\
+		PROG=$(SHELL);		\
+		SCRIPT=$(srcdir)/misc/units;	\
+	fi;	\
+	if ! test x$(SHELL_TEST) = x; then	\
+		SHELL_OPT=--shell=$(SHELL_TEST);	\
+	fi;							\
```

Comment:
When I implement shrinker in python, let's remove misc/units!

Reply:
The problem is that I have only implemented 5 actions: help, run, clean, tmain and clean-tmain.
Other actions (fuzz, shrink, noise, chop|slap, validate-input) are not implemented.

Also, I don't know how to install python3 on CentOS. So, `ci/circleci: centos_make` still uses `misc/units`.

Reply:
> The problem is that I have only implemented 5 actions: help, run, clean, tmain and clean-tmain.
Other actions (fuzz, shrink, noise, chop|slap, validate-input) are not implemented.

I see. Let's keep the shell script for awhile.

> Also, I don't know how to install python3 on CentOS. So, ci/circleci: centos_make still uses misc/units.

This is my job.  I will take a look the way to install python3 on CentOS.

Reply:
I've tried installing python3 to the environment established by docker.io/centos:latest.
What we have to do is quite simple: just adding python3 to yum command line like:

```
diff --git a/circle.yml b/circle.yml
index e143612a..db184682 100644
--- a/circle.yml
+++ b/circle.yml
@@ -40,7 +40,7 @@ jobs:
            name: Install build tools
            # TODO: enable spell checker
            command: |
-              yum -y --enablerepo=PowerTools install gcc automake autoconf pkgconfig make libxml2-devel jansson-devel libyaml-devel findutils diffutils
+              yum -y --enablerepo=PowerTools install python3 gcc automake autoconf pkgconfig make libxml2-devel jansson-devel libyaml-devel findutils diffutils
        - run:
            name: Build
            command: |
```

I hope this helps you.

Reply:
I confirmed that it worked.